libs/elasticsearch-nio/src/main/java/org/elasticsearch/nio/InboundChannelBuffer.java

```diff
@@ -65,15 +59,22 @@ public InboundChannelBuffer(Supplier<Page> pageSupplier) {
     public void close() {
         if (isClosed.compareAndSet(false, true)) {
             Page page;
-            List<RuntimeException> closingExceptions = new ArrayList<>();
+            RuntimeException closingException = null;
             while ((page = pages.pollFirst()) != null) {
                 try {
                     page.close();
                 } catch (RuntimeException e) {
-                    closingExceptions.add(e);
+                    if (closingException == null) {
+                        closingException = e;
+                    } else {
+                        closingException.addSuppressed(e);
+                    }
                 }
             }
-            ExceptionsHelper.rethrowAndSuppress(closingExceptions);
```

Comment:
Honestly I think we should just copy these helper classes 1 to 1 and then work on a common-lib jar. It's time for it anyway @rjernst @jasontedor WDYT?

Reply:
@s1monw I think it's probably time too, but I would want to use this opportunity to mark as much as possible in an internal package so that we are free to break whenever we want (obviously `ActionListener` would not be in such a package). When we go JDK 9 minimum, we can take this further.

Reply:
Can we separate the steps?

Reply:
@s1monw All I am suggesting is that when we move something (e.g., a utility class) that does not need to be part of the public API (because it's for example not used in the client or the plugin API) that we mark it internal. Nothing more. 😄